apps/web/lib/app-providers.tsx

```diff
@@ -40,14 +41,22 @@ const CustomI18nextProvider = (props: AppPropsWithChildren) => {
 
 const AppProviders = (props: AppPropsWithChildren) => {
   const session = trpc.useQuery(["viewer.session"]).data;
+  // No need to have intercom on public pages - Good for Page Performance
+  const isPublicPage = usePublicPage();
+  const RemainingProviders = (
+    <SessionProvider session={session || undefined}>
+      <CustomI18nextProvider {...props}>{props.children}</CustomI18nextProvider>
+    </SessionProvider>
+  );
+
   return (
     <TelemetryProvider value={createTelemetryClient()}>
       <IdProvider>
-        <DynamicIntercomProvider>
-          <SessionProvider session={session || undefined}>
-            <CustomI18nextProvider {...props}>{props.children}</CustomI18nextProvider>
-          </SessionProvider>
-        </DynamicIntercomProvider>
+        {isPublicPage ? (
+          RemainingProviders
```

Comment:
isn't this supposed to be `<RemainingProviders />`?

Reply:
No. It is not a component. It is just a variable storing the JSX.
Had it been a component like `()=>{return <SessionProvider></SessionProvider>}` it would be used like that.

Reply:
NIT: ah yeah i meant that, isn't it better to show as a component?

Reply:
Performance-wise, it is better I believe. Though, practically it won't make a significant difference in both cases.
Also, I find it better as I am just reusing JSX :)